src/components/header/header_links/header_links.tsx

```diff
@@ -17,83 +17,139 @@
  * under the License.
  */
 
-import React, { Component } from 'react';
+import React, {
+  HTMLAttributes,
+  FunctionComponent,
+  useState,
+  useEffect,
+} from 'react';
 import classNames from 'classnames';
 
-import { CommonProps } from '../../common';
-import { EuiIcon } from '../../icon';
-import { EuiPopover } from '../../popover';
+import { CommonProps, keysOf } from '../../common';
+import { EuiIcon, IconType } from '../../icon';
+import { EuiPopover, EuiPopoverProps } from '../../popover';
 import { EuiI18n } from '../../i18n';
-import { EuiHeaderSectionItemButton } from '../header_section';
+import {
+  EuiHeaderSectionItemButton,
+  EuiHeaderSectionItemButtonProps,
+} from '../header_section';
+import { EuiBreakpointSize } from '../../../services/breakpoint';
+import { EuiHideFor, EuiShowFor } from '../../responsive';
 
-interface State {
-  isOpen: boolean;
-}
+type EuiHeaderLinksGutterSize = 'xs' | 's' | 'm' | 'l';
+type EuiHeaderLinksPopoverButtonProps = EuiHeaderSectionItemButtonProps & {
+  iconType?: IconType;
+};
 
-export class EuiHeaderLinks extends Component<CommonProps, State> {
-  state: State = {
-    isOpen: false,
+export type EuiHeaderLinksProps = CommonProps &
+  HTMLAttributes<HTMLElement> & {
+    /**
+     * Spacing between direct children
+     */
+    gutterSize?: EuiHeaderLinksGutterSize;
+    /**
+     * A list of named breakpoints at which to show the popover version
+     */
+    popoverBreakpoints?: EuiBreakpointSize[] | 'all' | 'none';
+    /**
+     * Extend the functionality of the EuiPopover.button which is a EuiHeaderSectionItemButton.
+     * With the addition of `iconType` to change the display icon which defaults to `apps`
+     */
+    popoverButtonProps?: EuiHeaderLinksPopoverButtonProps;
+    /**
+     * Extend the functionality of the EuiPopover
+     */
+    popoverProps?: Omit<EuiPopoverProps, 'button' | 'closePopover'>;
   };
 
-  onMenuButtonClick = () => {
-    this.setState({
-      isOpen: !this.state.isOpen,
-    });
-  };
+const gutterSizeToClassNameMap: {
+  [gutterSize in EuiHeaderLinksGutterSize]: string;
+} = {
+  xs: '--gutterXS',
+  s: '--gutterS',
+  m: '--gutterM',
+  l: '--gutterL',
+};
+export const GUTTER_SIZES = keysOf(gutterSizeToClassNameMap);
 
-  closeMenu = () => {
-    this.setState({
-      isOpen: false,
-    });
+export const EuiHeaderLinks: FunctionComponent<EuiHeaderLinksProps> = ({
+  children,
+  className,
+  gutterSize = 's',
+  popoverBreakpoints = ['xs', 's'],
+  popoverButtonProps,
+  popoverProps,
+  ...rest
+}) => {
+  const { onClick: _onClick, iconType = 'apps', ...popoverButtonRest } = {
+    ...popoverButtonProps,
   };
 
-  componentDidMount() {
-    window.addEventListener('resize', this.closeMenu);
-  }
+  const [mobileMenuIsOpen, setMobileMenuIsOpen] = useState(false);
+
+  const onMenuButtonClick = (e: any) => {
```

Comment:
```suggestion
  const onMenuButtonClick: MouseEventHandler<HTMLButtonElement> = e => {
```

(and will need to import `MouseEventHandler` from react)